src/node_http_parser.cc

```diff
@@ -472,6 +472,16 @@ class Parser : public AsyncWrap {
   static void Consume(const FunctionCallbackInfo<Value>& args) {
     Parser* parser;
     ASSIGN_OR_RETURN_UNWRAP(&parser, args.Holder());
+
+    // Argument coming from JS land with unchecked type.
+    // Make sure it is a valid external object.
+    if (!args[0]->IsExternal()) {
+      parser->env()->isolate()->ThrowException(Exception::TypeError(
+        String::NewFromUtf8(parser->env()->isolate(),
+        "Object conversion of argument failed.")));
+      return;
+    }
```

Comment:
@gireeshpunathil FWIW, you can use a message with a CHECK assertion (though one may qualify it as a little hack). A common pattern used in some parts of the codebase is `CHECK(expr && "message")`.